workflowparsers/phonopy/calculator.py

```diff
@@ -93,15 +96,51 @@ def read_kpath(filename):
     return generate_kpath_parameters(points, [labels], npoints)
 
 
-def generate_kpath_ase(cell, symprec, logger=None):
+def test_non_canonical_hexagonal(cell: Cell, symprec: float) -> Optional[int]:
+    """
+    Tests if the cell is a non-canonical hexagonal cell
+    and returns the index of the ~ 60 degree angle (error range controlled by `symprec`).
+    """
     try:
-        lattice = aselattice.get_lattice_from_canonical_cell(Cell(cell))
+        target = 60
+        angles = cell.angles()
+        lattices = cell.lengths()
+    except AttributeError:
+        raise ValueError('Cell is not ase.cell.Cell')
+
+    # 2 tests:
+    ## 1. if there is only one angle close to 60 degrees
+    ## 2. if there is only one pair of lattice vectors with the same length
+    condition_angles = (angles > target - symprec) & (angles < target + symprec)
+    lattice_pairs = list(combinations(lattices, 2))
+    if (len(match_id := np.where(condition_angles)[0]) == 1) and (
+        sum([lat[1] - symprec <= lat[0] <= lat[1] + symprec for lat in lattice_pairs])
+        == 1
+    ):
+        return int(match_id[0])
+    return None
+
+
+def generate_kpath_ase(cell: Cell, symprec: float, logger=None) -> list:
+    try:
+        ase_cell = Cell(cell)
+        if isinstance(
+            rot_axis_id := test_non_canonical_hexagonal(ase_cell, 1e2 * symprec), int
+        ):  # be more lenient with the angle
+            logger.warning(
+                """Non-canonical hexagonal cell detected. Will correct the orientation."""
```

Comment:
Why 3 "?

Reply:
?

Reply:
There are three quotes. Loggers normally had 1.

Reply:
Since they're Python objects, they should be able to handle both.
You raise a good point, though: what is their behavior wrt the server consuming the log messages? You made me curious now, so I'll track this down.

Anyhow, here a single-line string is also fine, so I'll correct it.